### ISSUE TITLE
SpadeR from Anne Chao?

### ISSUE THREAD

**Issue by PhilippeJeanneret** [JohnsonHsieh/iNEXT#4]
Dear Johnson,
I am an intensive user of iNEXT and I was wondering if you could help me for SpadeR. For any reason, I could not download SpadeR from github of Anne, and could not install the package from the ZIP file I have downloaded.
Is there any problem with the ZIP package you may know about?

Thanks for your answer and the very useful programing of iNEXT.
Philippe Jeanneret

**Comment by JohnsonHsieh** [comment]
Hi Philippe, 
I think you could try to install SpadeR via devtools again 
```
install.packages('devtools')
library(devtools)
install_github('AnneChao/SpadeR')
library(SpadeR)
```
See detial for readme from https://github.com/AnneChao/SpadeR.

**Comment by PhilippeJeanneret** [comment]
Dear Johnson,
Thanks a lot for your quick answer. However, I could still not install SPADE via devtools, gettint the following message:
> library(devtools)
> install_github('AnneChao/SpadeR', lib= 'C:/Program Files/R/R-3.1.0/library')
> Downloading GitHub repo AnneChao/SpadeR@master
> Error in curl::curl_fetch_memory(url, handle = handle) : 
 > Timeout was reached
I also tried to install from the ZIP file without any success. I run a R 3.1.0 version. Could that be the reason of no success?
Thanks in advance for your help.
Philippe

**Comment by JohnsonHsieh** [comment]
First, I think you could try to add the `options` command as following 
```
options(download.file.method = "wininet")
library(devtools)
install_github('AnneChao/SpadeR', lib= 'C:/Program Files/R/R-3.1.0/library')
```

Alternative, 
Run cmd command
1. unzip SpadeR-master.zip
2. go to `SpadeR-master` directory, e.g. (d:/mypath/SpadeR-master)
```
cd d:
cd mypath/
```
3. install SpadeR source code version
```
R CMD INSTALL SpadeR-master
```

**Comment by PhilippeJeanneret** [comment]
Hi Johnson,
I could not install SpadeR even with option "wininet". I got the same error message, even with R version 3.2.2:
------------------------------------------------------------
> options(download.file.method = "wininet")
> library(devtools)
> install_github('AnneChao/SpadeR', lib= 'C:/Program Files/R/R-3.1.0/library')
Downloading GitHub repo AnneChao/SpadeR@master
Error in curl::curl_fetch_memory(url, handle = handle) : 
  Timeout was reached
---------------------------------------------------------------
With the CMD command, I got the following error message:
"Warning: invalid package 'SpadeR-master'
Error: ERROR: no packages specified".

Sorry to disturb you again. I hope you can find out where the trick is.
Thanks in advance.
Best wishes,
Philippe

**Comment by JohnsonHsieh** [comment]
This discussion would help you to fix this problem :
https://github.com/hadley/devtools/issues/877

**Comment by PhilippeJeanneret** [comment]
Hi Johnson,
Yes I looked at Hadley's advice but I am not sure I can use that way because of firewall issue, etc.
But instead of installing from github, why can I not correctly install from the ZIP file ?
In RStudio, there is no error message by doing "install.packages(".../SpadeR-master.zip, repos = NULL, type = "win.binary") but then the package is unzipped and a folder SpadeR-master is done, but seems not correctly installed. For instance, scripts are in "R" subfolder as "R File" and not as .RDB, or RDX.
From .ZIP, I could not re-install iNEXT (fortunately, I made a copy and can work with it now) but all other packages as ZIP were no problem.
Can you look at your ZIP files and tell me I have to proceed to install SpadeR from a ZIP file?

Thank you very much in advance.
Best wishes,
Philippe

**Comment by JohnsonHsieh** [comment]
Hi Philippe,
To install R package download from github repo, such as `SpadeR-master.zip`, `iNEXT-master.zip`,
You could not use `install.packages` function, you should use:
```
devtools::load_all("iNEXT")
```
Thx for your kindly report :)
Johnson

**Comment by PhilippeJeanneret** [comment]
Hi Johnson,
Thanks a lot Johnson, this worked! many many thanks for your patience!
So to summarize in case someone needs:
1) download the ZIP file from github AnneChao/SpadeR (or iNEXT) into the library folder of R,
2) unzip "SpadeR-master" into the library folder of R,
3) use 'devtools::load_all("path/to/SpadeR-master")'.
Notice: in RStudio, package "SpadeR" does not appear in the list of packages but the package is effectively loaded and works.

Thanks again and have a nice day.
Best wishes,
Philippe

**Comment by PhilippeJeanneret** [comment]
Dear Johnson,
I used SpadeR now and tried to calculate the number of shared species in two communities. There are things which seem strange.
When I have two communities like this:
> test <- data.frame(x = 1:20, y = c(0,0,0,0,0,2,4:17))
> ChaoShared(test, datatype="abundance", se=TRUE, nboot=200, conf=0.95)
I got that error message:
Error in rmultinom(B, n2, p2) : NA in probability vector 
Note that the community x has no 0 abundances (all y species are in x).
In contrast, if I do that:
> test <- data.frame(x = c(1:15,0,0,0,0,0), y = c(0,0,0,0,0,2,4:17))
>ChaoShared(test, datatype="abundance", se=TRUE, nboot=200, conf=0.95)
I got the normal results.
I have tested with several pairs of communities and my conclusion is if one community is a subset of the second one, i.e. one community has no 0, "ChaoShared" cannot calculate.
So my question: would it not be possible to improve the code so that "ChaoShared" can calculate in case one of the two communities has no 0 ?
Thank you in advance for your reply.
Best wishes,
Philippe
